### PR TITLE
Add documentation for geo replication

### PR DESCRIPTION
**Pull request by josephschorr**

### CODE REVIEW

File: quay-enterprise/geo-replication.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Georeplication of storage in Quay Enterprise
```
Comment (robszumski) [inline quay-enterprise/geo-replication.md]: mixed case. below is title case, this is sentence. I think the official style guide is sentence case

File: quay-enterprise/geo-replication.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+Georeplication allows for a single globally-distributed Quay Enterprise to serve container images from localized storage.
+
+When setup and enabled, all pushes performed will first save their image data to the preferred storage engine for the Quay Enterprise instance. Following push, the image data will be replicated in the background to other storage engines (configurable). Pulling of images will always make use of the closest *avaliable* storage engine, ensuring availability and performance.
```
Comment (robszumski) [inline quay-enterprise/geo-replication.md]: I haven't seen this term before: "storage engines". Could we state that as "storage locations"? Does that even make sense?

File: quay-enterprise/geo-replication.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+**NOTE: Ensure that there is a high availability storage engine (S3, GCS, RADOS, Swift) in each geographic region and that each region can access each storage engine. Local disk storage is not compatible with georeplication at this time.**
```
Comment (robszumski) [inline quay-enterprise/geo-replication.md]: This seems like more than just a "note". Can we restate it as a prereq or some sort of base knowledge?

File: quay-enterprise/geo-replication.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+Sign in to a super user account and visit `http://yourregister/superuser` to view the management panel:
+
+<img src="img/superuser.png" class="img-center" alt="Enterprise Registry Management Panel"/>
+
+## Enable Storage Replication
+
+1. Click the configuration tab (<span class="fa fa-gear"></span>) and scroll down to the section entitled <strong>Registry Storage</strong>.
+2. Click "Enable Storage Replication".
+3. Add each of the storage engines to which data will be replicated. All storages to be used must be listed and **all regions must have access to _all_ storage engines**.
```
Comment (robszumski) [inline quay-enterprise/geo-replication.md]: This seems quite important but I'm not sure if this sentence makes me 100% what I need to do. So my europe install needs to be able to access Swift in the US?

File: quay-enterprise/geo-replication.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+4. If global and universal replication is required, under each storage engine click "Replicate to storage engine by default". This will ensure that all images are replicated to that storage engine. To instead enable per-namespace replication, please contact support.
```
Comment (robszumski) [inline quay-enterprise/geo-replication.md]: global in this context means a few different things. Do you mean:

global as in applies to all repos?
global as in all geographic regions?

File: quay-enterprise/geo-replication.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+1. Copy the config.yaml to all machines running Quay Enterprise
+2. For each machine in each region, add a `QUAY_DISTRIBUTED_STORAGE_PREFERENCE` environment variable with the preferred storage engine for the region in which the machine is running.
+
+    For example, for a machine running in Europe:
+
+    ```
+    docker run -d -p 443:443 -p 80:80 -v /conf/stack:/conf/stack -e QUAY_DISTRIBUTED_STORAGE_PREFERENCE=europestorage quay.io/coreos/registry:versiontag
+    ```
+
+    *NOTE: The ID specified must match the name of a storage engine*
```
Comment (robszumski) [inline quay-enterprise/geo-replication.md]: which ID? I don't see an ID parameter

is there a list of all storage engines?

File: quay-enterprise/geo-replication.md (hunk 1 above)
Comment (josephschorr) [inline quay-enterprise/geo-replication.md]: Ok

File: quay-enterprise/geo-replication.md (hunk 2 above)
Comment (josephschorr) [inline quay-enterprise/geo-replication.md]: Not really, no. We use storage engines in the config tool and our code base

File: quay-enterprise/geo-replication.md (hunk 3 above)
Comment (josephschorr) [inline quay-enterprise/geo-replication.md]: Moved into its own section

File: quay-enterprise/geo-replication.md (hunk 4 above)
Comment (josephschorr) [inline quay-enterprise/geo-replication.md]: Removed and clarified above

File: quay-enterprise/geo-replication.md (hunk 5 above)
Comment (josephschorr) [inline quay-enterprise/geo-replication.md]: All repos

File: quay-enterprise/geo-replication.md (hunk 6 above)
Comment (josephschorr) [inline quay-enterprise/geo-replication.md]: Clarified

**Comment by josephschorr** [comment]
@robszumski: Ping!

**Comment by robszumski** [comment]
LGTM. Thoughts @joshix?

File: quay-enterprise/geo-replication.md (hunk 2 above)
Comment (joshix) [inline quay-enterprise/geo-replication.md]: When georeplication is configured, container image pushes will be written to the preferred storage engine for that QE instance. After the initial push, image data will be replicated in the background to other storage engines. This list of replication location is configurable. An image pull will always use the closest available storage engine, to maximize pull performance.

Then, for bonus points: define "closest"?

**Comment by joshix** [comment]
Just that one rewrite of a graf; otherwise LGTM

**Pull request merged**